scripts/dts/extract/globals.py

```diff
@@ -331,6 +331,56 @@ def build_cell_array(prop_array):
 
     return ret_array
 
+def child_to_parent_unmap(cell_parent, gpio_index):
+    # This function returns a (gpio-controller, pin number) tuple from
+    # cell_parent (identified as a 'nexus node', ie: has a 'gpio-map'
+    # property) and gpio_index.
+    # Note: Nexus nodes and gpio-map property are described in the
+    # upcoming (presumably v0.3) Device Tree specification, chapter
+    # 'Nexus nodes and Specifier Mapping'.
+
+    # First, retrieve gpio-map as a list
+    raw_map = reduced[cell_parent]['props']['gpio-map']
+
+    # We now need to build the map as an array of array, which will
+    # allow to use gpio_index
+
+    # map raws are encoded as follows:
+    # [child specifier][gpio controller phandle][parent specifier]
+    # child specifier field length is connector property #gpio-cells
+    # parent specifier field length is parent property #gpio-cells
+
+    child_specifier_size = reduced[cell_parent]['props']['#gpio-cells']
+
+    # Assumption 1: We assume parent #gpio-cells is constant across
+    # the map, so we take the value of the first occurrence and apply
+    # to the whole map.
+    parent = phandles[raw_map[child_specifier_size]]
+    parent_specifier_size = reduced[parent]['props']['#gpio-cells']
+    array_cell_size = child_specifier_size + 1 + parent_specifier_size
+
+    # Now that raws length is known, build the map as an array of arrays
+    index = 0
+    map_array = []
+    while index < len(raw_map):
+        map_array.append(raw_map[index:index+array_cell_size])
+        index += array_cell_size
+
+    # Return the gpio controller and pin using the provided index
+    if gpio_index not in [item[0] for item in map_array]:
+        # gpio_index not an index of map_array
+        return None, None
+    else:
+        # Return gpio_controller phandle and specifier pin
+        parent_controller_phandle = map_array[gpio_index][child_specifier_size]
```

Comment:
Is this looking up the right element?

Note that it's not comparing `gpio_index` against `item[0]` for each `item` in `map_array`, unlike the code above it.

Could maybe do something like this:

```python
for entry in gpio_map_entries:
    if entry[0] == gpio_index:
        parent_controller_phandle = entry[child_specifier_size]
        ...
        return phandles[...]

# gpio_index does not appear in gpio-map
return None, None
```

Reply:
Right, I kinda assumed first cell of each raw would be same as raw index.
This is indeed a shortcut, as it could be designed otherwise.

Reply:
Fixed